ihe-interop/src/main/java/org/motechproject/ihe/interop/domain/CdaTemplate.java

```diff
@@ -27,7 +29,25 @@ public CdaTemplate() {
     public CdaTemplate(String templateName, Byte[] templateData, Map<String, String> properties) {
         this.templateName = templateName;
         if (templateData != null) {
-            this.templateData = templateData.clone();
+
+            byte[] bytes = new byte[templateData.length];
```

Comment:
are you sure this works as expected ? As far as I know the CdaTemplate object is added via MDS, I'm not sure, however probably the new empty instance is created and then values are set separately, so this constructor will not be used. Generally adding this kind of parsing in constructors it's not the best idea